lib/suse/connect/api.rb

```diff
@@ -62,24 +62,32 @@ def list_subscriptions
 
       def send_bulk_system_update(systems, system_limit = nil)
         system_limit ||= BULK_SYSTEM_REQUEST_LIMIT
+        updated_systems = { systems: [] }
         systems.in_batches(of: system_limit) do |batched_systems|
           params = prepare_payload_for_bulk_update(batched_systems)
           response = make_single_request(
             :put,
             "#{connect_api}/organizations/systems",
             { body: params.to_json }
           )
-          yield response
+          updated_systems[:systems] = updated_systems[:systems].concat(response[:systems])
         end
       rescue RequestError => e
         # :nocov: TODO: https://github.com/SUSE/rmt/issues/911
         # change some params here and start the bulk update.
         if e.response.code == 413
+          @logger.info("Hit payload limit with: #{system_limit}")
           system_limit = e.response.headers['X-Payload-Entities-Max-Limit'].to_i
-          return send_bulk_system_update(systems, system_limit)
+          send_bulk_system_update(systems, system_limit)
         end
+      # rubocop:disable Lint/RescueException
+      rescue Exception => e
```

Comment:
What's the reason for introducing this `rescue` block? 
Afaik the only thing it's doing is writing an error log, but wouldn't the call stack also do that without catching it here? 
I'm asking because it needs a rubocop exception and is not tested.

Reply:
Ahh good point. There's a generic rescue in `scc.rb`. I will remove this.